lib/taskana-core-test/src/test/java/acceptance/taskcomment/query/QueryTaskCommentAccTest.java

```diff
@@ -0,0 +1,460 @@
+    }
+
+    @WithAccessId(user = ACCESS_USER_1_1_ID)
+    @Test
+    void should_ReturnListedValues_For_QueryColumnCreatorLongName() {
+      List<String> listedValues =
+          taskService
+              .createTaskCommentQuery()
+              .listValues(TaskCommentQueryColumnName.CREATOR_FULL_NAME, null);
+      assertThat(listedValues).hasSize(2);
+    }
+
+    @WithAccessId(user = ACCESS_USER_1_1_ID)
+    @Test
+    void should_ReturnListedValues_For_QueryColumnCreated() {
+      List<String> listedValues =
+          taskService.createTaskCommentQuery().listValues(TaskCommentQueryColumnName.CREATED, null);
+      assertThat(listedValues).hasSize(3);
+    }
+
+    @WithAccessId(user = ACCESS_USER_1_1_ID)
+    @Test
+    void should_ReturnListedValues_For_QueryColumnModified() {
+      List<String> listedValues =
+          taskService
+              .createTaskCommentQuery()
+              .listValues(TaskCommentQueryColumnName.MODIFIED, null);
+      assertThat(listedValues).hasSize(3);
+    }
+  }
+
+  @Nested
+  @TestInstance(Lifecycle.PER_CLASS)
+  class OffsetConstraintsTaskComments {
+    @WithAccessId(user = ACCESS_USER_1_1_ID)
+    @Test
+    void should_ConfirmQueryListOffset_When_ProvidingOffsetAndLimit() {
+      List<TaskComment> offsetAndLimitResult = taskService.createTaskCommentQuery().list(1, 2);
+      List<TaskComment> regularResult = taskService.createTaskCommentQuery().list();
+
+      assertThat(offsetAndLimitResult).hasSize(2);
+      assertThat(offsetAndLimitResult.get(0))
+          .isNotEqualTo(regularResult.get(0))
+          .isEqualTo(regularResult.get(1));
+    }
+
+    @WithAccessId(user = ACCESS_USER_1_1_ID)
+    @Test
+    void should_ReturnEmptyList_When_ProvidingWrongConstraints() {
+      List<TaskComment> result = taskService.createTaskCommentQuery().list(0, 1000);
+      assertThat(result).hasSize(3);
+
+      result = taskService.createTaskCommentQuery().list(100, 1000);
+      assertThat(result).isEmpty();
+    }
+  }
+
+  @Nested
+  @TestInstance(Lifecycle.PER_CLASS)
+  class SingleResultTaskComments {
+
+    @WithAccessId(user = ACCESS_USER_1_1_ID)
+    @Test
+    void should_ReturnSingleTaskComment_When_UsingSingleMethod() {
+      TaskComment single =
+          taskService
+              .createTaskCommentQuery()
+              .idIn(comment1.getId())
+              .creatorIn(ACCESS_USER_1_1_ID)
+              .single();
+
+      assertThat(single.getId()).isEqualTo(comment1.getId());
+    }
+
+    @WithAccessId(user = ACCESS_USER_1_1_ID)
+    @Test
+    void should_ThrowException_When_SingleMethodRetrievesMoreThanOneEventFromDatabase() {
+      TaskCommentQuery query = taskService.createTaskCommentQuery().creatorIn(ACCESS_USER_1_1_ID);
+      assertThatThrownBy(query::single).isInstanceOf(TooManyResultsException.class);
+    }
+  }
+
+  @Nested
+  @TestInstance(Lifecycle.PER_CLASS)
+  class TaskCreatorFullNameTaskCommentsWithAdditionalUserInfo
```

Comment:
The class name is not clear. What do you think about "CreatorFullNameWhenAdditionalUserInfoIsTrue"

Reply:
I have changed it to TaskCreatorFullNameWhenAdditionalUserInfoIsTrue

What do you think?

Reply:
Why TaskCreator? Doesn't the test check for the Creator of the TaskComment?

Reply:
Yes, you are right, I have changed them

Reply:
Changing it to ...CreatorFullNameOfTaskComment... might be misleading because of the FullNameOfTaskComment part. What do you think about "CreatorFullName" without TaskComment? The test is about TaskComments, so it should be clear that the TaskComment is meant

Reply:
I agree, I have changed it to "CreatorFullName"